client/components/signup-form/index.jsx

```diff
@@ -267,9 +284,9 @@ export default React.createClass( {
 						id="email"
 						name="email"
 						type="email"
-						value={ formState.getFieldValue( this.state.form, 'email' ) || this.props.email }
+						value={ formState.getFieldValue( this.state.form, 'email' ) }
 						isError={ formState.isFieldInvalid( this.state.form, 'email' ) }
-						isValid={ formState.isFieldValid( this.state.form, 'email' ) }
+						isValid={ this.state.validationInitialized && formState.isFieldValid( this.state.form, 'email' ) }
```

Comment:
In my testing, if we don't have that check, the field flashes green before properly validating.

You can test this by sendin an invite to an existing WP.com user and then trying to accept that invite while logged out.